libr/core/cmd_anal.c

```diff
@@ -7435,36 +7436,49 @@ static void cmd_anal_abt(RCore *core, const char *input) {
 			RList *path;
 			RListIter *pathi;
 			RListIter *bbi;
-			bool first_path = true, first_bb = true;
-			if (json) {
-				r_cons_printf ("[");
-			}
+			pj_a (pj);
 			r_list_foreach (paths, pathi, path) {
-				if (json) {
-					if (!first_path) {
-						r_cons_printf (", ");
-					}
+				pj_a (pj);
+				r_list_foreach (path, bbi, bb) {
+					pj_n (pj, bb->addr);
 				}
-				r_cons_printf ("[");
+				pj_end (pj);
+				r_list_purge (path);
+				free (path);
+			}
+			pj_end (pj);
+			r_cons_println (pj_string (pj));
+			pj_free (pj);
+			r_list_purge (paths);
+			free (paths);
+		}
+	}
+	break;
+	case ' ': {
+		ut64 addr;
+		char *p;
+		int n = 1;
+		p = strchr (input + 1, ' ');
+		if (p) {
+			*p = '\0';
+			n = *(++p)? r_num_math (core->num, p): 1;
+		}
+		addr = r_num_math (core->num, input + 1);
+		RList *paths = r_core_anal_graph_to (core, addr, n);
+		if (paths) {
+			RAnalBlock *bb;
+			RList *path;
+			RListIter *pathi;
+			RListIter *bbi;
+			r_list_foreach (paths, pathi, path) {
 				r_list_foreach (path, bbi, bb) {
-					if (json && !first_bb) {
-						r_cons_printf (", ");
-					}
 					r_cons_printf ("0x%08" PFMT64x, bb->addr);
```

Comment:
i dont get why you do r_cons_printf and then r_cons_newline, wen you can just add \n at the end of this printf